src/charm_state.py

```diff
@@ -65,16 +127,68 @@ def __init__(self, msg: str):
         self.msg = msg
 
 
+def _valid_storage_size_str(size: str) -> bool:
+    """Validate the storage size string.
+
+    Args:
+        size: Storage size string.
+
+    Return:
+        Whether the string is valid.
+    """
+    # Checks whether the string confirms to using the KiB, MiB, GiB, TiB, PiB,
+    # EiB suffix for storage size as specified in config.yaml.
+    valid_suffixes = ["KiB", "MiB", "GiB", "TiB", "PiB", "EiB"]
+    valid = any(suffix == size[-3:] for suffix in valid_suffixes)
```

Comment:
How about
```suggestion
    return size[-3:] in valid_suffixes and size[:-3].isdigit()
```

Reply:
Good idea. The following lines needs to be deleted as well.